File: Lv2/스택:큐 기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    cut = 1
    list = [math.ceil((100-progresses[i])/speeds[i]) for i in range(len(speeds))]
    day = list[0]
    for i in range(1,len(list)):
        if day >= list[i]:
            cut += 1
        else:
            answer.append(cut)
            cut = 1
            day = list[i]
        if i == len(list)-1:
            answer.append(cut)
        
    return answer

File: lv2 Swift/스택:큐 기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var day = ceil((100 - Double(progresses[0]))/Double(speeds[0]))
    var complete = 1
    var ans = [Int]()
    
    for i in 1..<speeds.count {
        var temp = ceil((100 - Double(progresses[i]))/Double(speeds[i]))
        
        if day < temp {
            ans.append(complete)
            
            day = ceil((100 - Double(progresses[i]))/Double(speeds[i]))
            complete = 1
            
        } else {complete += 1}
        
        if i == speeds.count-1 {ans.append(complete)}
    }
    
    return ans
}
